### PR TITLE
bump pytorch version to 2.1.0

### PR DESCRIPTION
**Pull request by johnnv1**
This patch adds torch 2.1.0 on CI to ensure support on it, skips the falling tests + the issues within Pytorch dynamo have been related, and fixes the new typing issues

### CODE REVIEW

File: test/filters/test_guided.py [hunk 1]
```diff
@@ -88,6 +89,9 @@ def test_module(self, shape, kernel_size, eps, subsample, device, dtype):
         op_module = GuidedBlur(kernel_size, eps, subsample=subsample)
         self.assert_close(op_module(guide, img), op(guide, img, kernel_size, eps, subsample=subsample))
 
+    @pytest.mark.skipif(
+        torch_version() == '2.1.0', reason="Failing with: Argument of Integer should be of numeric type, got s3 + 3."
```
Comment (johnnv1) [inline test/filters/test_guided.py:93]: I don't understand exactly the failing log. We should open an issue on Pytorch

File: test/geometry/transform/test_crop3d.py [hunk 2]
```diff
@@ -131,6 +132,10 @@ def test_gradcheck(self, device, dtype):
             kornia.geometry.transform.center_crop3d, (img, (3, 5, 7)), raise_exception=True, fast_mode=True
         )
 
+    @pytest.mark.skipif(
+        torch_version() == '2.1.0',
+        reason="unsupported operand type(s) for @: 'FakeTensor' and 'FakeTensor' on `normalize_homography3d`",
```
Comment (johnnv1) [inline test/geometry/transform/test_crop3d.py:137]: We should open an issue on Pytorch reporting it

Comment (johnnv1) [inline test/geometry/transform/test_crop3d.py:137]:
```suggestion
        reason="https://github.com/pytorch/pytorch/issues/110680:  unsupported operand type(s) for @: 'FakeTensor' and 'FakeTensor' on `normalize_homography3d`",
```

File: test/filters/test_guided.py (hunk 1 above)
Comment (johnnv1) [reply test/filters/test_guided.py:93]: https://github.com/pytorch/pytorch/issues/110696

Comment (johnnv1) [inline test/filters/test_guided.py:93]:
```suggestion
        torch_version() == '2.1.0', reason="https://github.com/pytorch/pytorch/issues/110696 - Failing with: Argument of Integer should be of numeric type, got s3 + 3."
```

**Review by edgarriba — APPROVED** [review]

**Pull request merged**